src/Functions/text/searchAnyAll.cpp

```diff
@@ -169,8 +169,33 @@ ColumnPtr FunctionSearchImpl<SearchTraits>::executeImpl(
     else
         throw Exception(ErrorCodes::BAD_ARGUMENTS, "Function '{}' supports only tokenizers 'default', 'ngram', 'split', and 'no_op'", name);
 
-    const auto & col_needles_tokens = col_needles->getDataAt(0);
-    std::vector<String> needles = DefaultTokenExtractor().getTokens(col_needles_tokens.data, col_needles_tokens.size);
+    std::vector<String> needles;
+    {
+        const Array & needles_array = [&col_needles]
+        {
+            if (const ColumnConst * col_needles_const = checkAndGetColumnConst<ColumnArray>(col_needles.get()))
+                return col_needles_const->getValue<Array>();
+            else if (const ColumnArray * col_needles_non_const = checkAndGetColumn<ColumnArray>(col_needles.get()))
```

Comment:
Because of l. 38 (`isColumnConst`), the case that the needes array is non-const rejected already earlier (but . We can remove l. 178-183.